lib/view_component/with_content_helper.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module ViewComponent
+  module WithContentHelper
+    def with_content(value = nil, &block)
```

Comment:
`content=` gives me the impression that the content will already be evaluated, which is not the case with `with_content`.

I think the main intention of `with_content` is telling the component: "hold this content and evaluate it when you have access to a view_context".

That's why we can do `MyComponent.new.with_content(OtherComponent.new)` without a `view_context` available, since `OtherComponent` isn't rendered when `with_content` is called, only when we render the `MyComponent` instance.

As for some other points:

> Allow us to avoid the complexity of supporting blocks

Actually, we don't need to support blocks afaik 🤔 

> The one downside is that we lose some amount of composability

I think composability is a great part of the API, since it can allow us to do:

```rb
component = MyComponent.new.content(
  OtherComponent.new.with_content(
    AnotherComponent.new...
```

Reply:
I agree with @manuelpuyol here. That being said, we _could_ start with just the string API and add support for passing in components later? I'm fine with either approach.

Reply:
> I think composability is a great part of the API, since it can allow us to do:
> 
> ```ruby
> component = MyComponent.new.content(
>   OtherComponent.new.with_content(
>     AnotherComponent.new...
> ```

I agree, but at the same time I see it as us re-implementing some part of Rails view `render` API (at least when we support renderables).

I could also see something like the following if we wanted to pre-render content for components.

```ruby
# alternative:  some kind of block wrapper
component = ViewComponent.render(MyComponent.new) do |c|
  render OtherComponent.new do
    AnotherComponent.new
  end
end
```

I think that, combined with `with_content` (imagining that it only supports strings) could be a powerful combination.

```ruby
# with some kind of block wrapper
component = ViewComponent.render(MyComponent.new) do |c| # provides a controllerless view_context to render
  render OtherComponent.new do
    AnotherComponent.new.with_content(content_tag(:h1, "hello!")) # with_content is helpful for injecting pre-rendered or Ruby generated content
  end
end
```

Reply:
I think that pivoting to string only is a great idea, we can mature the API and have further discussions (like this one) as we see more use-cases in production!

Reply:
removed the Component implementation in https://github.com/github/view_component/pull/875/commits/963dd774375a0cadba8eda033f6d96981d0a00bd